client/my-sites/checkout/checkout/source-payment-box.jsx

```diff
@@ -0,0 +1,231 @@
+
+		this.setState( {
+			formDisabled: submitState.disabled
+		} );
+	}
+
+	paymentMethodByType( paymentType ) {
+		if ( paymentType === 'ideal' ) {
+			return 'WPCOM_Billing_Stripe_Source_Ideal';
+		}
+		return 'WPCOM_Billing_Stripe_Source';
+	}
+
+	redirectToPayment( event ) {
+		const origin = this.getLocationOrigin( location );
+		event.preventDefault();
+
+		this.setSubmitState( {
+			info: translate( 'Setting up your %(paymentProvider)s payment', {
+				args: { paymentProvider: this.getPaymentProviderName() } } ),
+			disabled: true
+		} );
+
+		let cancelUrl = origin + '/checkout/';
+
+		if ( this.props.selectedSite ) {
+			cancelUrl += this.props.selectedSite.slug;
+		} else {
+			cancelUrl += 'no-site';
+		}
+
+		const dataForApi = {
+			payment: assign( {}, this.state, {
+				paymentMethod: this.paymentMethodByType( this.props.paymentType ),
+				successUrl: origin + this.props.redirectTo(),
+				cancelUrl,
+			} ),
+			cart: this.props.cart,
+			domainDetails: this.props.transaction.domainDetails
+		};
+
+		// get the redirect URL from rest endpoint
+		wpcom.undocumented().transactions( 'POST', dataForApi, function( error, result ) {
+			let errorMessage;
+			if ( error ) {
+				if ( error.message ) {
+					errorMessage = error.message;
+				} else {
+					errorMessage = translate( "We've encountered a problem. Please try again later." );
+				}
+
+				this.setSubmitState( {
+					error: errorMessage,
+					disabled: false
+				} );
+			} else if ( result.redirect_url ) {
+				this.setSubmitState( {
+					info: translate( 'Redirecting you to your bank to complete the payment.' ),
+					disabled: true
+				} );
+				analytics.ga.recordEvent( 'Upgrades', 'Clicked Checkout With Source Payment Button' );
+				analytics.tracks.recordEvent( 'calypso_checkout_with_source_' + this.props.paymentType );
+				location.href = result.redirect_url;
+			}
+		}.bind( this ) );
+	}
+
+	renderButtonText() {
+		if ( cartValues.cartItems.hasRenewalItem( this.props.cart ) ) {
+			return translate( 'Purchase %(price)s subscription with %(paymentProvider)s', {
+				args: { price: this.props.cart.total_cost_display, paymentProvider: this.getPaymentProviderName() }
+			} );
+		}
+
+		return translate( 'Pay %(price)s with %(paymentProvider)s', {
+			args: { price: this.props.cart.total_cost_display, paymentProvider: this.getPaymentProviderName() }
+		} );
+	}
+
+	renderBankOptions() {
+		// Source https://stripe.com/docs/sources/ideal
+		const idealBanks = {
+			abn_amro: 'ABN AMRO',
+			asn_bank: 'ASN Bank',
+			bunq: 'Bunq',
+			ing: 'ING',
+			knab: 'Knab',
+			rabobank: 'Rabobank',
+			regiobank: 'RegioBank',
+			sns_bank: 'SNS Bank',
+			triodos_bank: 'Triodos Bank',
+			van_lanschot: 'Van Lanschot',
+		};
+
+		const idealBanksOptions = map( idealBanks, ( text, optionValue ) => (
+			<option value={ optionValue } key={ optionValue }>{ text }</option>
+		) );
+
+		return [
+			<option value="" key="-">{ translate( 'Please select your bank…' ) }</option>,
```

Comment:
Hi! I've found a possible matching string that has already been translated **1** time:
`translate( 'Please select your bank.' )` *ES Score: 15*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).